libraries/chain/include/eosio/chain/block_header_state.hpp

```diff
@@ -89,13 +89,8 @@ struct block_header_state {
    finalizer_policy_ptr                active_finalizer_policy; // finalizer set + threshold + generation, supports `digest()`
    proposer_policy_ptr                 active_proposer_policy;  // producer authority schedule, supports `digest()`
 
-   // block time when proposer_policy will become active
-   // current algorithm only two entries possible, for the next,next round and one for block round after that
-   // The active time is the next,next producer round. For example,
-   //   round A [1,2,..12], next_round B [1,2,..12], next_next_round C [1,2,..12], D [1,2,..12]
-   //   If proposed in A1, A2, .. A12 becomes active in C1
-   //   If proposed in B1, B2, .. B12 becomes active in D1
-   flat_map<block_timestamp_type, proposer_policy_ptr>     proposer_policies;
+   std::optional<proposer_policy_ptr> latest_proposed_proposer_policy;
+   std::optional<proposer_policy_ptr> latest_pending_proposer_policy;
```

Comment:
Can you add the rules 1 to 4 from the issue here as comments, so that we don't have to lookup the issue to understand what the rules are.